cpp/src/gandiva/precompiled/string_ops.cc

```diff
@@ -84,6 +83,15 @@ VAR_LEN_OP_TYPES(BINARY_RELATIONAL, greater_than_or_equal_to, >=)
   INNER(NAME, utf8)                \
   INNER(NAME, binary)
 
+int to_binary_from_hex(char ch) {
```

Comment:
inline..is this only to handle ascii characters?

Reply:
will handle only valid hexadecimal characters
a-f,  A-F,  0 - 9

There is a check too before invoking this method _isxdigit(hd1) && isxdigit(hd2)_